### DISCUSSION TITLE
Deny filetransfer between RDP- and local session - xrdp 0.9.12-1

### DISCUSSION THREAD

**Discussion by McSh4dow** [neutrinolabs/xrdp#2252] [Q&A]
Hi!

I am using Ubuntu 20.04 with xrdp 0.9.12-1 as a terminal server.

I'd like to deny users copying files from the terminal server to their local system - text clipboard needs work. I found out xrdp 0.9.15+ supports blocking filetransfer through the "EnableFuseMount=false" setting in /etc/xrdp/sesman.ini.

Right now 0.9.15+ is only for Ubuntu 21.10+ in the official repo.

Is there a workaround for this? Or do I need to install 0.9.15+ on Ubuntu 20.04?

Thanks and best wishes

**Comment by matt335672** [comment]
In `/etc/xrdp/xrdp.ini`, under `[Channels]`, try setting `rdpdr=false`. That should disable the file system redirector.

**Reply by McSh4dow** [reply]
Hi! Unfortunately it still works in both directions after I set this and rebooted. Is there something else I can try?

**Reply by matt335672** [reply]
Are you talking about clipboard file transfers or remote drive file transfers?

I assumed you meant the latter - disabling rdpdr should fix those. Clipboard file transfers on that version of xrdp can only be disabled by disabling the cliprdr channel as well. That will mean you have no clipboard at all with the Xorg backend. You'll get a clipboard with the Xvnc backend but it's quite flaky on that version.

**Reply by McSh4dow** [reply]
I'm talking about clipboard file transfer. Is there an easy way to install xrdp 0.9.15+ on Ubuntu 20.04?

**Reply by matt335672** [reply]
Well, it depends what you mean by 'easy'.

There are scripts out there which will install contemporary versions of xrdp, but in general the authors of those scripts (unlike most distros) choose not to engage with this project directly. Consequently, as an upstream project, we are not in a position to endorse or recommend that approach - which isn't to say that it wouldn't work for you of course. That depends on your exact requirements and use-case(s).

The Debian/Ubuntu packages are patched a bit to add extra security and to make sure xrdp runs smoothly on those operating systems. The best result you could probably get would be to learn how to build Debian packages from source, and then download the xrdp package sources for 22.04 LTS (from [here](https://packages.ubuntu.com/jammy/xrdp)) and rebuild on 20.04. That will probably provide a better alternative than installing xrdp yourself from source.

Hope that's a bit useful - I'm aware it's not the best reply you could have had.

**Reply by McSh4dow** [reply]
Actually this might help out a lot. I will try this one on Monday and add another reply when I'm facing any further problems.

Installing it from source did not work perfectly - somehow I disabled the whole clipboard. Maybe because I did not compile Fuse during the installation at all? I will check it out as well.

Thank you!
